src/main/java/com/internetitem/logback/elasticsearch/writer/ElasticsearchWriter.java

```diff
@@ -86,15 +108,29 @@ public void sendData() throws IOException {
 			urlConnection.disconnect();
 		}
 
+		updateBuffersOnSuccess();
+	}
+
+	public void clear() {
+		updateBuffersOnSuccess();
+	}
+
+	private void updateBuffersOnSuccess() {
 		sendBuffer.setLength(0);
-		if (bufferExceeded) {
-			errorReporter.logInfo("Send queue cleared - log messages will no longer be lost");
+		if (backupBuffer.length() > 0) {
```

Comment:
Ok, so we sent out the sendBuffer, so we add the backupBuffer to the sendBuffer, and reset the exceed flags as needed. Buffers are (mostly) emptied, we're back in business.